code/modules/antagonists/eldritch_cult/knowledge/flesh_lore.dm

```diff
@@ -125,19 +141,33 @@
 	human_target.become_husk()
 	human_target.apply_status_effect(/datum/status_effect/ghoul)
 	human_target.faction |= "heretics"
+
+	RegisterSignal(human_target,COMSIG_MOB_DEATH, .proc/remove_ghoul)
+	RegisterSignal(human_target,COMSIG_HERETIC_REMOVE_GHOUL,.proc/remove_ghoul)
```

Comment:
Why not just call remove_ghoul directly from the point of deconversion? It would be better in this case since nothing else will ever use the signal COMSIG_HERETIC_REMOVE_GHOUL